sdks/python/apache_beam/examples/inference/pytorch_sentiment_streaming.py

```diff
@@ -0,0 +1,222 @@
+      help='Pub/Sub subscription to read from')
+  parser.add_argument(
+      '--project', default='apache-beam-testing', help='GCP project ID')
+  return parser.parse_known_args(argv)
+
+
+def ensure_pubsub_resources(
+    project: str, topic_path: str, subscription_path: str):
+  """Ensures that the required Pub/Sub topic and subscription exist."""
+  publisher = pubsub_v1.PublisherClient()
+  subscriber = pubsub_v1.SubscriberClient()
+
+  topic_name = topic_path.split("/")[-1]
+  subscription_name = subscription_path.split("/")[-1]
+
+  full_topic_path = publisher.topic_path(project, topic_name)
+  full_subscription_path = subscriber.subscription_path(
+      project, subscription_name)
+
+  try:
+    publisher.get_topic(request={"topic": full_topic_path})
+  except Exception:
+    publisher.create_topic(name=full_topic_path)
+
+  try:
+    subscriber.get_subscription(
+        request={"subscription": full_subscription_path})
+  except Exception:
+    subscriber.create_subscription(
+        name=full_subscription_path, topic=full_topic_path)
+
+
+def cleanup_pubsub_resources(
+    project: str, topic_path: str, subscription_path: str):
+  """Deletes Pub/Sub topic and subscription created for the pipeline."""
+  publisher = pubsub_v1.PublisherClient()
+  subscriber = pubsub_v1.SubscriberClient()
+
+  topic_name = topic_path.split("/")[-1]
+  subscription_name = subscription_path.split("/")[-1]
+
+  full_topic_path = publisher.topic_path(project, topic_name)
+  full_subscription_path = subscriber.subscription_path(
+      project, subscription_name)
+
+  try:
+    subscriber.delete_subscription(
+        request={"subscription": full_subscription_path})
+    print(f"Deleted subscription: {subscription_name}")
+  except Exception as e:
+    print(f"Failed to delete subscription: {e}")
+
+  try:
+    publisher.delete_topic(request={"topic": full_topic_path})
+    print(f"Deleted topic: {topic_name}")
+  except Exception as e:
+    print(f"Failed to delete topic: {e}")
+
+
+def run(
+    argv=None, save_main_session=True, test_pipeline=None) -> PipelineResult:
+  known_args, pipeline_args = parse_known_args(argv)
+  pipeline_options = PipelineOptions(pipeline_args)
+  pipeline_options.view_as(SetupOptions).save_main_session = save_main_session
+  pipeline_options.view_as(StandardOptions).streaming = True
+
+  model_handler = PytorchModelHandlerKeyedTensor(
+      model_class=DistilBertForSequenceClassification,
+      model_params={'config': DistilBertConfig(num_labels=2)},
+      state_dict_path=known_args.model_state_dict_path)
+  ensure_pubsub_resources(
+      project=known_args.project,
+      topic_path=known_args.pubsub_topic,
+      subscription_path=known_args.pubsub_subscription)
+
+  tokenizer = DistilBertTokenizerFast.from_pretrained(known_args.model_path)
+
+  pipeline = test_pipeline or beam.Pipeline(options=pipeline_options)
+
+  # 1. Load data pipeline: read lines from GCS file and send to Pub/Sub
```

Comment:
I think we need to add a few things to this:

1) These should run as separate pipelines and we should only monitor the results of the streaming portion
2) We need a way of applying a consistent rate to the input elements in pub/sub. A better approach might be to either (a) just do this independently via a script, or (b) within Dataflow use periodic impulse or state/timers to control the rate at which we emit. This will likely work better as a helper which we can use across pipelines
3) For streaming pipelines, we should enable autoscaling

Reply:
Added autoscaling.
Can we merge this PR with the current approach (it is the easiest way to run two pipelines simultaneously).
Then we can implement reusable independent script that controls input rate.

Reply:
I think we should at least handle (1) and run these as two pipelines. Otherwise, I do not think we are measuring a meaningful dataset